uaclient/actions.py

```diff
@@ -51,26 +56,49 @@ def attach_with_token(
     """
     from uaclient.timer.update_messaging import update_motd_messages
 
+    contract_client = contract.UAContractClient(cfg)
+    attached_at = datetime.datetime.now(tz=datetime.timezone.utc)
+
     try:
-        contract.request_updated_contract(
```

Comment:
Just double checking, can't we update `request_updated_contract` with this extra logic ?

Reply:
Yes, and that could solve the specific need of this PR, but the refactor stemmed from two observations:
1. The exception cases below with comments claiming to run after a partial attach are wrong - they run after failed attaches as well. That is due to the attachment and enablement being combined in the one request_updated_contract function and that function doesn't separate the exceptions raised in attachment vs during enablement (my first idea of refactoring was do exactly that).
2. The request_updated_contract function was kind-of two-functions-in-one. Most of it's uses were as "refresh", and one of its uses was as "attach". Since there was only one usage of the "attach" variant of request_updated_contract, I thought i made sense to bring the attach logic up into our "attach_with_token" function. That leaves only the refresh code behind, which is what every other caller was using.

The other thing is that I think `attach_with_token` should be the place where "attach"y things happen, and it mostly is, so it would be confusing to put some actions related to "attach" (saving attachment date, and sending machine activity ping) down in request_updated_contract.